source/backend/api/Helpers/Extensions/PrincipalExtensions.cs

```diff
@@ -11,18 +11,28 @@ public static class PrincipalExtensions
     {
         public static void ThrowInvalidAccessToAcquisitionFile(this ClaimsPrincipal principal, IUserRepository userRepository, IAcquisitionFileRepository acquisitionFileRepository, long acquisitionFileId)
         {
-            if (principal is null)
-            {
-                throw new ArgumentNullException(nameof(principal));
-            }
+            ArgumentNullException.ThrowIfNull(principal);
 
             var pimsUser = userRepository.GetUserInfoByKeycloakUserId(principal.GetUserKey());
-            var acquisitionFile = acquisitionFileRepository.GetById(acquisitionFileId);
+            PimsAcquisitionFile acquisitionFile = acquisitionFileRepository.GetById(acquisitionFileId);
 
             if (pimsUser?.IsContractor == true && !acquisitionFile.PimsAcquisitionFileTeams.Any(x => x.PersonId == pimsUser.PersonId))
             {
                 throw new NotAuthorizedException("Contractor is not assigned to the Acquisition File's team");
             }
         }
+
+        public static void ThrowInvalidAccessToDispositionFile(this ClaimsPrincipal principal, IUserRepository userRepository, IDispositionFileRepository dispositionFileRepository, long dispostionFileId)
+        {
+            ArgumentNullException.ThrowIfNull(principal);
+
+            var pimsUser = userRepository.GetUserInfoByKeycloakUserId(principal.GetUserKey());
+            PimsDispositionFile dispostionFile = dispositionFileRepository.GetById(dispostionFileId);
+
+            if (pimsUser?.IsContractor == true && !dispostionFile.PimsDispositionFileTeams.Any(x => x.PersonId == pimsUser.PersonId))
+            {
+                throw new NotAuthorizedException("Contractor is not assigned to the Disposition File's team");
```

Comment:
This seems better, if a contractor does not have access to a file they should see an error not an override.

Reply:
all good